File: src/main/java/com/github/bjoernpetersen/jmusicbot/playback/QueueChangeListener.java
package com.github.bjoernpetersen.jmusicbot.playback;

import javax.annotation.Nonnull;

public interface QueueChangeListener {

  void onAdd(@Nonnull QueueEntry entry);

  void onRemove(@Nonnull QueueEntry entry);

  void onMove(@Nonnull QueueEntry entry, int fromIndex, int toIndex);
}

File: src/main/kotlin/com/github/bjoernpetersen/musicbot/spi/player/QueueChangeListener.kt
package com.github.bjoernpetersen.musicbot.spi.player

import com.github.bjoernpetersen.musicbot.api.player.QueueEntry

interface QueueChangeListener {
    fun onAdd(entry: QueueEntry)
    fun onRemove(entry: QueueEntry)
    fun onMove(entry: QueueEntry, fromIndex: Int, toIndex: Int)
}
